Makefile

```diff
@@ -27,15 +27,25 @@ define color
   @printf '\033[0;34m==== $1 ====\033[0m\n'
 endef
 
-all: bin/ekam-bootstrap deps
+all: bin/ekam
+
+bin/ekam: bin/ekam-bootstrap deps
 	$(call color,building ekam with ekam)
 	@rm -f bin/ekam
 	@CXX="$(CXX)" CXXFLAGS="-std=c++14 $(CXXFLAGS) -pthread" LIBS="-pthread" bin/ekam-bootstrap -j$(PARALLEL)
 	@test -e bin/ekam && printf "=====================================================\nSUCCESS\nOutput is at bin/ekam\n=====================================================\n"
 
-continuous: bin/ekam-bootstrap deps
+# NOTE: uses the newly built ekam instead of the bootstrap ekam so that LSP is available
+continuous: bin/ekam
```

Comment:
This is awkward because `continuous` itself will overwrite `bin/ekam`. Also if you are building from scratch, using `make continuous` means ekam will be built three times: bootstrap, then a full build using ekam, and then a second full build using ekam in continuous mode.

It might make more sense to say that when using `make continuous`, you must provide an existing external ekam binary...

Reply:
Hmm, that makes sense to me. I changed it to require an external binary.